litemapy/schematic.py

```diff
@@ -278,6 +278,80 @@ def _tonbt(self):
 
         return root
 
+    def to_structure_nbt(self, mc_version=MC_DATA_VERSION, gzipped=True, byteorder='big'):
+        structure = nbtlib.File(gzipped=gzipped, byteorder=byteorder)
+
+        structure['size'] = List[Int]([self.__width, self.__height, self.__length])
```

Comment:
A region's dimension may be negative. You need to store their absolute value and convert the coordinates you use accordingly.

Reply:
True. Fixed it.